tests/kafkatest/tests/streams/streams_multiple_rolling_upgrade_test.py

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from ducktape.mark import matrix
+from ducktape.mark.resource import cluster
+from kafkatest.services.streams import StreamsSmokeTestDriverService, StreamsSmokeTestJobRunnerService
+from kafkatest.tests.streams.base_streams_test import BaseStreamsTest
+from kafkatest.version import LATEST_0_10_2, LATEST_0_11_0, LATEST_1_0,  DEV_BRANCH, KafkaVersion
+
+
+class StreamsMultipleRollingUpgradeTest(BaseStreamsTest):
+    """
+     This test will verify a rolling upgrade of multiple streams
+     applications against all versions of streams against a single
+     broker version.
+
+     As new releases come out, just update the streams_upgrade_versions array to have the latest version
+     included in the list.
+
+     A prerequisite for this test to succeed
+     is the inclusion of all parametrized versions of kafka in kafka/vagrant/base.sh
+     (search for get_kafka()).
+     As new versions are released the kafka/tests/kafkatest/version.py file
+     needs to be updated as well.
+
+     You can find what's been uploaded to S3 with the following command
+
+     aws s3api list-objects --bucket kafka-packages --query 'Contents[].{Key:Key}
+    """
+    # adding new version to this list will cover broker and streams version
+    streams_upgrade_versions = [str(LATEST_0_10_2), str(LATEST_0_11_0), str(LATEST_1_0), str(DEV_BRANCH)]
```

Comment:
what about 0.10.1 and 1.1 ?

Reply:
Added `0.10.1` I was thinking `0.10.2` was the first version we could do a rolling upgrade from.  I've left out `1.1.0` for the moment as I was thinking this PR will be merged before the 1.1 release.

Reply:
I see. That makes sense. I run into the same issue and put a workaround in #4636 -- we should clean this up, after we put proper testing for older versions into place as discussed in #4636.